### PR TITLE
Feature/extra budget module

### PR DESCRIPTION
**Pull request by EstebanVU**
- New extra budget module was created.
- New logic to create a full copy of the data that is currently in the original budget.

### CODE REVIEW

**Review by KuroScar42 — APPROVED** [review]

File: src/components/views/ProjectDetail/ExtraBudget/BudgetSubcontract/BudgetSubcontract.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+  const tableHeader: TTableHeader[] = [
+    { name: 'name', value: appStrings.name },
+    { name: 'quantity', value: appStrings.quantity },
+    { name: 'cost', value: appStrings.cost, isGreen: true },
+    { name: 'subtotal', value: appStrings.subtotal, isGreen: true },
+  ];
+
+  const getSubcontracts = async () => {
+    const successCallback = (response: IBudgetSubcontract[]) =>
```
Comment (Chinaski1979) [inline src/components/views/ProjectDetail/ExtraBudget/BudgetSubcontract/BudgetSubcontract.tsx:52]: Please in this scenario try use one line functions when possible, is not necessary to indent on to the next line

File: src/components/views/ProjectDetail/ExtraBudget/BudgetSubcontract/BudgetSubcontract.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  const validationSchema = yup.object().shape({
+    name: yup.string().required(appStrings?.requiredField),
+    quantity: yup.number().positive().required(appStrings?.requiredField),
+    cost: yup.number().positive().required(appStrings?.requiredField),
+  });
+
+  useEffect(() => {
+    let abortController = new AbortController();
+    getSubcontracts();
+    return () => {
```
Comment (Chinaski1979) [inline src/components/views/ProjectDetail/ExtraBudget/BudgetSubcontract/BudgetSubcontract.tsx:109]: return () => abortController.abort();

File: src/components/views/ProjectDetail/ExtraBudget/BudgetSummary/BudgetSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    const grandTotal = totalDirectCost + adminFee;
+    return { totalDirectCost, adminFee, grandTotal };
+  };
+
+  useEffect(() => {
+    setBudgetTotals({ ...budget, ...calcTotals(budget) });
+  }, [budget]);
+
+  return budgetTotals ? (
+    <>
```
Comment (Chinaski1979) [inline src/components/views/ProjectDetail/ExtraBudget/BudgetSummary/BudgetSummary.tsx:44]: Unnecessary fragment.
https://reactjs.org/docs/fragments.html

**Review by Chinaski1979 — APPROVED** [review]
left a few comments. Please address the ones that apply. Ingore the others. Merge after updating.

**Pull request merged**